src/sels_fe/src/components/quiz/Quiz/QuizWithQuestionsCreate.jsx

```diff
@@ -0,0 +1,305 @@
+import { useDispatch, useSelector } from "react-redux"
+import { useHistory } from "react-router"
+
+import {
+  postQuestion,
+  postQuiz,
+  updateQuestion,
+  updateQuiz,
+} from "../../../store/actions"
+
+import { Stepper, StepButton, Step, Box } from "@mui/material"
+import { IconButton, Button, Typography } from "@material-ui/core"
+
+import { Delete } from "@material-ui/icons"
+import { Divider, Stack } from "@mui/material"
+
+import QuizForm from "./components/QuizForm"
+import QuestionsForm from "./../Question/components/QuestionForm"
+import ChoiceForm from "./../Choice/components/ChoiceForm"
+import AnswersSelect from "../Answer/components/AnswersSelect"
+
+const steps = [
+  "Add Quiz Details",
+  "Add Questions",
+  "Add Choices",
+  "Select Answers",
+]
+let currentQuiz = null
+let currentQuestions = null
+
+function QuizWithQuestionsCreate() {
+  const quizzes = useSelector((state) => state.Quiz.quiz_list)
+  const question_list = useSelector((state) => state.Question.question_list)
+  const choices = useSelector((state) => state.Choice.choice_list)
+  let errorQuiz = useSelector((state) => state.Quiz.error)
+
+  const [activeStep, setActiveStep] = useState(0)
+  const [completed, setCompleted] = useState([])
+  const [update, setUpdate] = useState(false)
+
+  const [quiz, setQuiz] = useState({ name: "", description: "" })
+  const [questions, setQuestions] = useState([
+    { question: "", description: "", quiz: [] },
+  ])
+  const [questionAnswer, setQuestionAnswer] = useState([
+    { choice: 0, question: 0 },
+  ])
+
+  const dispatch = useDispatch()
+  const history = useHistory()
+
+  const handleQuizChange = (event) => {
+    const { name, value } = event.target
+    setQuiz({ ...quiz, [name]: value })
+  }
+
+  const handleQuestionsChange = (index, event) => {
+    let questionsFormVal = [...questions]
+    questionsFormVal[index][event.target.name] = event.target.value
+    setQuestions(questionsFormVal)
+  }
+
+  let addQuestionsForm = () => {
+    setQuestions([...questions, { question: "", description: "", quiz: [] }])
+  }
+
+  let removeQuestionsForm = (index) => {
+    let questionsFormValue = [...questions]
+    questionsFormValue.splice(index, 1)
+    setQuestions(questionsFormValue)
+  }
+
+  let handleSetQuestionAnswer = (event) => {
+    const { name, value } = event.target
+    setQuestionAnswer({ ...questionAnswer, [name]: value })
+  }
+
+  const handleSubmit = () => {
+    if (activeStep === 0) {
+      completed[0] !== true
+        ? dispatch(postQuiz(quiz))
+        : dispatch(updateQuiz(currentQuiz[0]?.id, quiz))
+    }
+    if (activeStep === 1) {
+      questions.map((question, index) => {
+        const question_id = new Array()
+        question_id.push(currentQuiz[0]?.id)
+        question["quiz"] = question_id
+        completed[1] !== true
+          ? dispatch(postQuestion(question))
+          : dispatch(updateQuestion(currentQuestions[index]?.id, question))
+      })
```

Comment:
To pass multiple objects, map the question array first.
Then pass to data at once
_Note: in below sample code, I did not consider `updateQuestion`_

```suggestion
      const data = questions.map((question, index) => {
        const question_id = new Array()
        question_id.push(currentQuiz[0]?.id)
        question["quiz"] = question_id
        return question
      })

      if (completed[1] !== true) {
        dispatch(postQuestion(data))
      }
```